python/ray/experimental/state.py

```diff
@@ -594,31 +626,141 @@ def micros_rel(ts):
                         "cat": "execute",
                         "pid": "Node " + str(worker["node_ip_address"]),
                         "tid": info["worker_id"],
-                        "id": str(worker),
+                        "id": str(task_id),
                         "ts": micros_rel(info["execute_start"]),
                         "ph": "X",
                         "name": info["function_name"] + ":execute",
-                        "args": delta_info,
+                        "args": total_info,
                         "dur": micros(info["execute_end"] -
-                                      info["execute_start"])
+                                      info["execute_start"]),
+                        "cname": "rail_animation"
                     }
                     full_trace.append(execute_trace)
+
             else:
+                if parent_info:
+                    parent_worker = workers[parent_info["worker_id"]]
+                    parent_times = self._get_times(parent_info)
+                    parent = {
+                        "cat": "submit_task",
+                        "pid": "Node " + str(parent_worker["node_ip_address"]),
+                        "tid": parent_info["worker_id"],
+                        "ts": micros_rel(task_profiles[task_table[task_id]
+                                                       ["TaskSpec"]
+                                                       ["ParentTaskID"]]
+                                          ["get_arguments_start"]),
+                        "ph": "s",
+                        "name": "SubmitTask",
+                        "args": {},
+                        "id": str(worker) + str(micros(min(parent_times)))
```

Comment:
Ok, I see it's a dictionary returned by `self.workers()`, but why are we converting that to a string and putting it in the json? What is the ID field here being used for?